### PR TITLE
fix(track): Send decisions for all experiments using an event when using track

### PR DESCRIPTION
**Pull request by aliabbasrizvi**

### CODE REVIEW

**Comment by aliabbasrizvi** [comment]
This includes commits from #135

File: optimizely/event_builder.py [hunk 1]
```diff
@@ -243,40 +243,39 @@ def _get_required_params_for_conversion(self, event_key, event_tags, decisions):
...
+      numeric_value = event_tag_utils.get_numeric_value(event_tags, self.config.logger)
+      if numeric_value is not None:
+        event_dict[event_tag_utils.NUMERIC_METRIC_TYPE] = numeric_value
 
-        snapshot[self.EventParams.EVENTS] = [event_dict]
+      if len(event_tags) > 0:
+        event_dict[self.EventParams.TAGS] = event_tags
 
-        return snapshot
+    snapshot.setdefault(self.EventParams.EVENTS, []).append(event_dict)
```
Comment (mikeng13) [inline optimizely/event_builder.py]: Do we need to setdefault as opposed to just initializing an empty array with that `event_dict`?

File: tests/base.py [hunk 2]
```diff
@@ -375,5 +375,152 @@ def setUp(self):
       }]
     }
 
-    self.optimizely = optimizely.Optimizely(json.dumps(self.config_dict))
+    self.config_dict_with_multiple_experiments = {
```
Comment (mikeng13) [inline tests/base.py]: Shouldn't this be passed in for that one unit test that uses it?

Comment (aliabbasrizvi) [reply tests/base.py]: I think it is ok

File: optimizely/event_builder.py (hunk 1 above)
Comment (aliabbasrizvi) [reply optimizely/event_builder.py]: Yeah, let me update that.

**Comment by aliabbasrizvi** [comment]
build-e2e

**Pull request merged**